tests/test_expectation_maximization.py

```diff
@@ -0,0 +1,173 @@
+"""Test bayesian expectation maximization process."""
+
+import numpy as np
+import pytest
+
+from reconstructSPI.iterative_refinement import expectation_maximization as em
+
+
+@pytest.fixture
+def n():
+    """Get test array size value for consistency."""
+    return 2
+
+
+@pytest.fixture
+def test_ir(n):
+    """Instantiate IterativeRefinement class for testing."""
+    ex_ctf = {
+        "s": np.ones((n, n)),
+        "a": np.ones((n, n)),
+        "def1": 1.0,
+        "def2": 1.0,
+        "angast": 0.1,
+        "kv": 0.1,
+        "cs": 0.1,
+        "bf": 0.1,
+        "lp": 0.1,
+    }
+    map_3d = np.zeros((n, n, n))
+    particles = np.zeros((n, n, n))
+    ctf_info = [
+        ex_ctf,
+    ] * n
+    itr = 2
+    ir = em.IterativeRefinement(map_3d, particles, ctf_info, itr)
+    return ir
+
+
+def test_split_array(test_ir, n):
+    """Test splitting of array in two halves."""
+    arr = np.zeros(n)
+    arr1, arr2 = test_ir.split_array(arr)
+    assert arr1.shape == (n // 2,)
+    assert arr2.shape == (n // 2,)
+
+    arr = np.zeros(n + 1)
+    arr1, arr2 = test_ir.split_array(arr)
+    assert arr1.shape == ((n + 1) // 2,)
+    assert arr2.shape == ((n + 1) // 2,)
+
+    arr = ["a", "b"]
+    arr1, arr2 = test_ir.split_array(arr)
+    assert len(arr1) == 1
+    assert len(arr2) == 1
+
+
+def test_fft_3d(test_ir, n):
+    """Test 3D fourier transform."""
+    arr = np.zeros((n, n, n))
+    fft_arr = test_ir.fft_3d(arr)
+    assert fft_arr.shape == arr.shape
+
+
+def test_ifft_3d(test_ir, n):
+    """Test 3D inverse fourier transform."""
+    fft_arr = np.zeros((n, n, n))
+    arr = test_ir.fft_3d(fft_arr)
+    assert fft_arr.shape == arr.shape
```

Comment:
These tests will be moved to compSPI once there is a working 3D FFT to go along with the 2D FFTs that are already there.

Reply:
Keep an eye on https://github.com/compSPI/compSPI/issues/33